pkg/api/v1/machine.go

```diff
@@ -42,7 +42,10 @@ type RegisterMachineRequest struct {
 }
 
 func (g *MachineGroup) RegisterMachine(ctx echo.Context) error {
-	_, _ = ctx.(*auth.HttpAuthContext)
+	cc, _ := ctx.(*auth.HttpAuthContext)
+	if (cc.AuthInfo.Token.TokenType != types.TokenTypeMachine) && (cc.AuthInfo.Token.TokenType != types.TokenTypeWorker) {
```

Comment:
Try replacing this with WithClusterAdmiAuth

Reply:
I can't here because I'm checking against a different token type